R/archive.R

```diff
@@ -278,26 +246,9 @@ epi_archive <-
       initialize = function(x, geo_type, time_type, other_keys,
                             additional_metadata, compactify,
                             clobberable_versions_start, versions_end) {
-        # Check that we have a data frame
-        if (!is.data.frame(x)) {
-          Abort("`x` must be a data frame.")
-        }
-
-        # Check that we have geo_value, time_value, version columns
-        if (!("geo_value" %in% names(x))) {
-          Abort("`x` must contain a `geo_value` column.")
-        }
-        if (!("time_value" %in% names(x))) {
-          Abort("`x` must contain a `time_value` column.")
-        }
-        if (!("version" %in% names(x))) {
-          Abort("`x` must contain a `version` column.")
-        }
-        if (anyNA(x$version)) {
-          Abort("`x$version` must not contain `NA`s",
-            class = "epiprocess__version_values_must_not_be_na"
-          )
-        }
+        assert_data_frame(x)
+        assert_subset(c("geo_value", "time_value", "version"), names(x))
+        assert(!anyNaN(x$version))
```

Comment:
I think this last one is wrong (should be `anyNA()`).


Generally, I'm not sure I like the messages produced by this package. There's of course a balance between offloading this stuff to well-tested functions and getting everything I want, but these seem too low-tech. For example:

``` r
library(checkmate)
x <- 1:10
names(x) <- letters[1:10]
assert_subset(c("geo_value", "time_value", "version"), names(x))
#> Error: Assertion on 'c("geo_value", "time_value", "version")' failed: Must be a subset of {'a','b','c','d','e','f','g','h','i','j'}, but has additional elements {'geo_value','time_value','version'}.

assert_data_frame(x)
#> Error: Assertion on 'x' failed: Must be of type 'data.frame', not 'integer'.
version <- c(1:5, NA)
assert(!anyNaN(version))
assert(!anyNA(version))
#> Error: Assertion on 'anyNA(version)' failed: FALSE.
```

The messages aren't wrapped or formatted as they were before. And the messages aren't nearly as clear. It's worth examining https://style.tidyverse.org/error-messages.html

Reply:
> should be `anyNA()`

Ah yea, in the idioms of this packages, it's `anyMissing()`. I can change that.

> these seem too low-tech
> the messages aren't nearly as clear

Seems like a matter of preference, so it's hard for me to know what you don't like about these. Custom error messages can always be done like this
```r
if (test_subset(c("geo_value", "time_value", "version"), names(x))) cli_abort("custom message")
```
Like you said, the off-loading to well tested functions is exactly what I like here. It's wild to me how much of our unit testing is the same "did this input type error correctly throw?" checks over and over. To me that's just downstream of how squirmy type checking can get in R. This package gives me type check tests I can trust, which frees up energy for testing the higher level logic of the package.

Reply:
Maybe better then to refactor the `arg_is_*()` type functions as you did in epipredict to use `{checkmate}`. That way you mostly get better messaging. 

In this particular case, to better mimic what the original messages were saying (though with better formatting), I'd like to see something like:
``` r
hardhat::validate_column_names(data.frame(geo_value = 1:5, time_value = 1:5), c("geo_value", "time_value", "version"))
#> Error in `hardhat::validate_column_names()`:
#> ! The following required columns are missing: 'version'.
```

<sup>Created on 2024-02-01 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
```